src/python/bot/tasks/corpus_pruning_task.py

```diff
@@ -471,25 +456,23 @@ def run(self, initial_corpus_path, minimized_corpus_path, bad_units_path):
 
     # Execute fuzzer with arguments for corpus pruning.
     logs.log('Running merge...')
-    result = self.runner.merge(
-        [minimized_corpus_path, initial_corpus_path],
-        CORPUS_PRUNING_TIMEOUT,
-        artifact_prefix=bad_units_path,
-        tmp_dir=self.context.merge_tmp_dir,
-        additional_args=additional_args)
-
-    # Sanity check that we didn't time out.
-    symbolized_output = stack_symbolizer.symbolize_stacktrace(result.output)
-    if result.timed_out:
+    try:
+      result = self.runner.minimize_corpus(
+          additional_args, [initial_corpus_path], minimized_corpus_path,
+          bad_units_path, CORPUS_PRUNING_TIMEOUT)
+    except engine.TimeoutError as e:
       raise CorpusPruningException(
-          'Corpus pruning timed out while merging corpus: %s.' %
-          symbolized_output)
-    # Sanity check that we didn't error out and there are files in minimized
-    # corpus after merging.
-    if (result.return_code or
-        not shell.get_directory_file_count(minimized_corpus_path)):
+          'Corpus pruning timed out while merging corpus: ' + e.message)
+    except engine.MergeError as e:
+      raise CorpusPruningException('Corpus pruning failed to merge corpus: ' +
+                                   e.message)
+
+    # Sanity check that there are files in minimized corpus after merging.
+    symbolized_output = stack_symbolizer.symbolize_stacktrace(result.logs)
```

Comment:
nit: this line doesn't relate to the comment line above, maybe move it one line higher

Reply:
done.